chimney-macro-commons/src/main/scala-3/io/scalaland/chimney/internal/compiletime/datatypes/SealedHierarchiesPlatform.scala

```diff
@@ -29,10 +29,14 @@ trait SealedHierarchiesPlatform extends SealedHierarchies { this: DefinitionsPla
         else if sym.flags.is(Flags.Module) then List(sym.typeRef.typeSymbol.moduleClass)
         else List(sym)
 
+      val order = Ordering.Option(Ordering.fromLessThan[Position] { (a, b) =>
+        a.startLine < b.startLine || (a.startLine == b.startLine && a.startColumn < b.startColumn)
+      })
```

Comment:
Crap... it's supposed to work but in `compileErrors` check it [throws an `AssertionError`](https://github.com/scalalandio/chimney/actions/runs/9252556722/job/25451767651?pr=543) - IMHO it's a bug and `pos` should not return `Some(position)` if in its internal representation `Span` is `NoSpam` (so there is _no_ `start`/`end`, not even 0).

I'll report it later, but for now I think we are forced to do something like:

```
_.pos.filter(pos => scala.util.Try(pos.start).isSuccess)
```
to work around this bug :/